server/neptune/workflows/internal/deploy/terraform/state.go

```diff
@@ -68,22 +69,34 @@ func (n *StateReceiver) Receive(ctx workflow.Context, c workflow.ReceiveChannel,
 		len(workflowState.Apply.OnWaitingActions.Actions) > 0 {
 		queuedDeployments := n.Queue.GetOrderedMergedItems()
 		revisionsSummary := n.Queue.GetQueuedRevisionsSummary()
+		runState := github.CheckRunQueued
+		var actions []github.CheckRunAction
 		var summary string
 
 		if workflowState.Apply.Status == state.WaitingJobStatus {
 			runLink := github.BuildRunURLMarkdown(deploymentInfo.Repo.GetFullName(), deploymentInfo.Commit.Revision, deploymentInfo.CheckRunID)
 			summary = fmt.Sprintf("This deploy is queued pending action on run for revision %s.\n%s", runLink, revisionsSummary)
 		} else if workflowState.Apply.Status == state.RejectedJobStatus || workflowState.Apply.Status == state.InProgressJobStatus {
 			// If the current deployment is Rejected or In Progress status, we need to restore the queued check runs to reflect that the queued deployments are not blocked.
-			summary = "This deploy is queued and will be processed as soon as possible.\n" + revisionsSummary
+			// If the queue is currently locked we need to provide the unlock action.
+			queueLock := n.Queue.GetLockState()
+			if queueLock.Status == lock.LockedStatus {
+				actions = append(actions, github.CreateUnlockAction())
+				runState = github.CheckRunActionRequired
+				revisionLink := github.BuildRevisionURLMarkdown(deploymentInfo.Repo.GetFullName(), queueLock.Revision)
+				summary = fmt.Sprintf("This deploy is locked from a manual deployment for revision %s.  Unlock to proceed.\n%s", revisionLink, revisionsSummary)
+			} else {
+				summary = "This deploy is queued and will be processed as soon as possible.\n" + revisionsSummary
+			}
```

Comment:
When we update checkruns for queued commits (after the blocking commit gets confirmed or rejected), we need to check if the queue was previously locked by something else and restore the unlock button for each commit.